hail/python/test/hailtop/hailctl/config/test_cli.py

```diff
@@ -20,40 +20,90 @@ def test_config_list_empty_config(runner: CliRunner):
 @pytest.mark.parametrize(
     'name,value',
     [
-        ('batch/backend', 'batch'),
+        ('domain', 'azure.hail.is'),
+        ('gcs_requester_pays/project', 'hail-vdc'),
+        ('gcs_requester_pays/buckets', 'hail,foo'),
+        ('batch/backend', 'service'),
         ('batch/billing_project', 'test'),
+        ('batch/regions', 'us-central1,us-east1'),
         ('batch/remote_tmpdir', 'gs://foo/bar'),
         ('query/backend', 'spark'),
-
-        # hailctl currently accepts arbitrary settings
-        ('foo/bar', 'baz'),
+        ('query/jar_url', 'gs://foo/bar.jar'),
+        ('query/batch_driver_cores', '1'),
+        ('query/batch_worker_cores', '1'),
+        ('query/batch_driver_memory', '1Gi'),
+        ('query/batch_worker_memory', 'standard'),
+        ('query/name_prefix', 'foo'),
+        ('query/disable_progress_bar', '1'),
     ],
 )
-def test_config_set(name: str, value: str, runner: CliRunner):
+def test_config_set_get_list_unset(name: str, value: str, runner: CliRunner):
     runner.invoke(cli.app, ['set', name, value], catch_exceptions=False)
 
     res = runner.invoke(cli.app, 'list', catch_exceptions=False)
     assert res.exit_code == 0
+    if '/' not in name:
+        name = f'global/{name}'
     assert res.stdout.strip() == f'{name}={value}'
 
     res = runner.invoke(cli.app, ['get', name], catch_exceptions=False)
     assert res.exit_code == 0
     assert res.stdout.strip() == value
 
+    res = runner.invoke(cli.app, ['unset', name], catch_exceptions=False)
+    assert res.exit_code == 0
 
-def test_config_get_bad_names(runner: CliRunner):
-    res = runner.invoke(cli.app, ['get', 'foo'], catch_exceptions=False)
+    res = runner.invoke(cli.app, 'list', catch_exceptions=False)
     assert res.exit_code == 0
     assert res.stdout.strip() == ''
 
-    res = runner.invoke(cli.app, ['get', '/a/b/c'], catch_exceptions=False)
+
+# backwards compatibility
+def test_config_get_unknown_names(bc_runner: CliRunner):
+    res = bc_runner.invoke(cli.app, ['get', 'email'], catch_exceptions=False)
+    assert res.exit_code == 0
+    assert res.stdout.strip() == 'johndoe@gmail.com'
+
+    res = bc_runner.invoke(cli.app, ['get', 'batch/foo'], catch_exceptions=False)
```

Comment:
This is a good catch and I think we should proceed in this PR as you have it currently. I feel like at some point though we should somehow communicate to users that they have bogus settings in their config. Maybe something we can discuss at team meeting?

Reply:
This could be part of a future `hailctl config check` where we also check the remote_tmpdir is correct for the regions and not cold storage.